testing/test_regressions.py

```diff
@@ -97,6 +100,52 @@ def test_case_mismatch_on_windows_git(tmp_path: Path) -> None:
     assert res is not None
 
 
+def test_fallback_message(wd: WorkDir, monkeypatch: pytest.MonkeyPatch) -> None:
+    """Check that no warning/error messages are normally outputted"""
+    # setup a basic project that would not trigger any (other) warning/errors
+    pyproject = """\
+        [build-system]
+        requires = ["setuptools", "setuptools_scm"]
+        build-backend = "setuptools.build_meta"
+
+        [project]
+        name = "example"
+        dynamic = ["version"]
+
+        [tool.setuptools_scm]
+        """
+    git_archival = """\
+        node: $Format:%H$
+        node-date: $Format:%cI$
+        describe-name: $Format:%(describe:tags=true,match=*[0-9]*)$
+        """
+    wd.write("pyproject.toml", textwrap.dedent(pyproject))
+    wd.write(".git_archival.txt", textwrap.dedent(git_archival))
+    wd.write(".gitattributes", ".git_archival.txt  export-subst")
+    wd.write("README.md", "")
+    wd("git init")
+    wd("git config user.email user@host")
+    wd("git config user.name user")
+    wd.add_command = "git add ."
+    wd.commit_command = "git commit -m test-{reason}"
+    wd.add_and_commit()
+    wd("git tag v0.1.0")
+    version = wd.get_version()
+    assert version == "0.1.0"
+
+    # Run build and check messages
+    monkeypatch.delenv("SETUPTOOLS_SCM_DEBUG")
+    # with caplog.at_level(logging.WARN):
+    res_build = subprocess.run(
+        [sys.executable, "-m", "build", "-nx"],
+        cwd=wd.cwd,
+        capture_output=True,
+    )
+    assert res_build.returncode == 0
+    # Maybe there is a cleaner way to use caplog?
+    assert not res_build.stderr
```

Comment:
Is there a better way to run and check this?

Reply:
A hack that might be reasonable would be a fake object directly passed to the integration function

Reply:
You mean instead of `python3 -m build`? I don't think that will end up well because what we want to test here is how `build` creates the sdist and from the sdist creates the wheel, and in both steps `setuptools-scm` is called with different "sources".

Another option I was thinking is calling [`build.__main__.main`](https://github.com/pypa/build/blob/0665ea731db6f431fb4f800b7f6849597c28677c/src/build/__main__.py#L391C5-L391C9) directly? Then all of the context should be preserved and the `caplog` should in principle work?

Reply:
as far as i remember, build has to start sub-processes for a build
so currently  have no idea for a clean solution